src/interface.jl

```diff
@@ -0,0 +1,664 @@
+    # we extract `data` so that we don't returne a `Tracked` type
+    return Tracker.data(Tracker.jacobian(b, y))
+end
+
+# TODO: allow batch-computation, especially for univariate case?
+"Computes the absolute determinant of the Jacobian of the inverse-transformation."
+logabsdetjac(b::ADBijector, x::Real) = log(abs(jacobian(b, x)))
+function logabsdetjac(b::ADBijector, x::AbstractVector{<:Real})
+    fact = lu(jacobian(b, x), check=false)
+    return issuccess(fact) ? log(abs(det(fact))) : -Inf # TODO: do this or not?
+end
+
+"""
+    logabsdetjacinv(b::Bijector, y)
+
+Just an alias for `logabsdetjac(inv(b), y)`.
+"""
+logabsdetjacinv(b::Bijector, y) = logabsdetjac(inv(b), y)
+
+###############
+# Composition #
+###############
+
+"""
+    ∘(b1::Bijector, b2::Bijector)
+    compose(ts::Bijector...)
+
+A `Bijector` representing composition of bijectors.
+
+# Examples
+It's important to note that `∘` does what is expected mathematically, which means that the
+bijectors are applied to the input right-to-left, e.g. first applying `b2` and then `b1`:
+```
+(b1 ∘ b2)(x) == b1(b2(x))     # => true
+```
+But in the `Composed` struct itself, we store the bijectors left-to-right, so that
+```
+cb1 = b1 ∘ b2                  # => Composed.ts == (b2, b1)
+cb2 = compose(b2, b1)          # => Composed.ts == (b2, b1)
+cb1(x) == cb2(x) == b1(b2(x))  # => true
+```
+"""
+struct Composed{A} <: Bijector
+    ts::A
+end
+
+compose(ts::Bijector...) = Composed(ts)
+
+# The transformation of `Composed` applies functions left-to-right
+# but in mathematics we usually go from right-to-left; this reversal ensures that
+# when we use the mathematical composition ∘ we get the expected behavior.
+# TODO: change behavior of `transform` of `Composed`?
+∘(b1::Bijector, b2::Bijector) = compose(b2, b1)
+
+inv(ct::Composed) = Composed(map(inv, reverse(ct.ts)))
+
+# # TODO: should arrays also be using recursive implementation instead?
+function (cb::Composed{<: AbstractArray{<: Bijector}})(x)
+    res = x
+    for b ∈ cb.ts
+        res = b(res)
+    end
+
+    return res
+end
+
+# recursive implementation like this allows type-inference
+_transform(x, b1::Bijector, b2::Bijector) = b2(b1(x))
+_transform(x, b::Bijector, bs::Bijector...) = _transform(b(x), bs...)
+(cb::Composed{<: Tuple})(x) = _transform(x, cb.ts...)
+
+function _logabsdetjac(x, b1::Bijector, b2::Bijector)
+    res = forward(b1, x)
+    return logabsdetjac(b2, res.rv) + res.logabsdetjac
+end
+function _logabsdetjac(x, b1::Bijector, bs::Bijector...)
+    res = forward(b1, x)
+    return _logabsdetjac(res.rv, bs...) + res.logabsdetjac
+end
+logabsdetjac(cb::Composed, x) = _logabsdetjac(x, cb.ts...)
+
+# recursive implementation of `forward`
+function _forward(f, b1::Bijector, b2::Bijector)
+    f1 = forward(b1, f.rv)
+    f2 = forward(b2, f1.rv)
+    return (rv=f2.rv, logabsdetjac=f2.logabsdetjac + f1.logabsdetjac + f.logabsdetjac)
```

Comment:
Since we initialize `f.logabsdetjac` to 0, this gives an error when we send a batch of samples. Please make it `.+` .